tools/lint.py

```diff
@@ -2,6 +2,7 @@
 import os
 import sys
 import subprocess
+import packaging.version
```

Comment:
Don't we vendor `_pep440` in `scipy._lib` for the purpose of avoiding this extra dependency? I mean, I guess this is just for devs instead of end users, but seems like we could still avoid the extra dependency in this case too.

Reply:
though I guess if SciPy isn't built yet that might get ugly

Reply:
ah okay, that looks like what I was looking for 👍

Reply:
hmm, maybe not though, because of the chicken and egg problem with SciPy not build yet, not sure how ugly that might get!

Reply:
I checked and `packaging` is transitively installed by `mamba env create -f environment.yml`. Given that trying to uninstall it wants to also uninstall `meson-python`, I think we're fine?

Reply:
Yes, that should be fine to rely on. `packaging` is the only dependency of `pyproject-metadata`, and that will remain a dependency of `meson-python` for the foreseeable future.